src/ripple/app/ledger/LedgerTiming.h

```diff
@@ -52,67 +56,74 @@ std::uint32_t roundCloseTime (
 // they should not be changed arbitrarily.
 
 // The percentage threshold above which we can declare consensus.
-int const minimumConsensusPercentage = 80;
+auto constexpr minimumConsensusPercentage = 80;
 
+using namespace std::chrono_literals;
 // All possible close time resolutions. Values should not be duplicated.
-int const ledgerPossibleTimeResolutions[] = { 10, 20, 30, 60, 90, 120 };
+std::chrono::seconds constexpr ledgerPossibleTimeResolutions[] =
+    { 10s, 20s, 30s, 60s, 90s, 120s };
 
+#ifndef _MSC_VER
 // Initial resolution of ledger close time.
-int const ledgerDefaultTimeResolution = ledgerPossibleTimeResolutions[2];
+auto constexpr ledgerDefaultTimeResolution = ledgerPossibleTimeResolutions[2];
+#else
+// HH Remove this workaround of a VS bug when possible
+auto constexpr ledgerDefaultTimeResolution = 30s;
```

Comment:
This is probably overkill, but for just a bit of extra safety, can we do something like:

`static_assert (ledgerDefaultTimeResolution == ledgerPossibleTimeResolutions[2], "Changing the default time resolution?")`